File: integration_tests/mockApis/alertsApi.ts
import { SuperAgentRequest } from 'superagent'
import { stubFor } from './wiremock'

const stubGetEscapeProfile = ({
  offenderNo,
  alertCode,
}: {
  offenderNo: string
  alertCode: string
}): SuperAgentRequest =>
  stubFor({
    request: {
      method: 'GET',
      url: `/alerts-api/prisoners/${offenderNo}/alerts?isActive=true&alertCode=XER,XEL,XELH`,
    },
    response: {
      status: 200,
      headers: { 'Content-Type': 'application/json;charset=UTF-8' },
      jsonBody: {
        content: [
          {
            alertCode: {
              code: alertCode,
            },
            activeFrom: '2016-09-14',
          },
        ],
      },
    },
  })

const stubAlertsApiPing = (statusCode = 200): SuperAgentRequest =>
  stubFor({
    request: {
      method: 'GET',
      urlPattern: `/alerts-api/health/ping`,
    },
    response: {
      status: statusCode,
      headers: { 'Content-Type': 'application/json;charset=UTF-8' },
      jsonBody: {
        status: statusCode,
        response: {},
      },
    },
  })

export default {
  stubGetEscapeProfile,
  stubAlertsApiPing,
}

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/AlertsApi.java
package uk.gov.justice.digital.hmpps.cattool.mockapis;

import com.github.tomakehurst.wiremock.junit.WireMockRule;
import groovy.json.JsonOutput;

import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
import static com.github.tomakehurst.wiremock.client.WireMock.get;

public class AlertsApi extends WireMockRule {

  public AlertsApi() {
    super(8089);
  }

  public void stubGetAlerts(String offenderno, boolean onEscapeList, boolean activeOnEscapeList) {
    this.stubFor(get("/prisoners/" + offenderno + "/alerts?isActive=true&alertCode=XER,XEL,XELH")
      .willReturn(aResponse()
        .withStatus(200)
        .withHeader("Content-Type", "application/json")
        .withBody(JsonOutput.toJson("{content: [{alertCode: {code: \"XER\"}, activeFrom: \"2025-01-01\"}, {alertCode: {code: \"XEL\"}, activeFrom: \"2025-01-01\"}]}"))
      ));
  }
}
